include/etcpal/thread.h

```diff
@@ -169,6 +169,10 @@ etcpal_error_t etcpal_thread_join(etcpal_thread_t* id);
 void etcpal_thread_sleep(unsigned int sleep_ms);
 #endif
 
+#if defined(ETCPAL_THREAD_JOIN_CAN_TIMEOUT) || DOXYGEN
```

Comment:
The typical EtcPal philosophy has been to not remove things from the public API based on compile definitions. So for example, on Linux you still have `etcpal_mutex_timed_lock()`, but the time parameter is ignored. `ETCPAL_MUTEX_HAS_TIMED_LOCK` is just informative - it tells you whether the parameter will be honored or not. I think we should do the same here - e.g. don't protect this function behind the #if.

Reply:
On the other platforms, the function can be implemented as a stub that just calls `etcpal_thread_join()`.